### DISCUSSION TITLE
How do Submit a nmea string to a NmeaDevice

### DISCUSSION THREAD

**Discussion by Skyhikeeper** [dotMorten/NmeaParser#108] [General]
I have a device that I receive Nmea string on, I need to pass them onto the parser to handle the strings.
How do i setup and use a streamdevice?

Terence

**Comment by dotMorten** [comment]
What sort of device? How is it connected?

**Comment by Skyhikeeper** [comment]
It is connected via a tcp port.

**Comment by Skyhikeeper** [comment]
Which receives a list of nmea strings which should be passed to the nmea parser to handle what to do with them.

**Reply by dotMorten** [reply]
Your tcp client should provide you with the stream. Just give hand that stream off to nmea device and it’ll take care of reading the strings from the stream and parsing it

**Comment by Skyhikeeper** [comment]
I am hosting the TcpListiner and receiving the strings on that.

should I allocate the stream to StreamDevice ?
